modules/webclient/src/main/scala/scaladex/client/Client.scala

```diff
@@ -36,27 +42,30 @@ object Client {
         .map(t => headers + ("Authorization" -> s"bearer $t"))
         .getOrElse(headers)
 
-    val root = s"https://github.com/$organization/$repository"
-    def base(v: String) = s"$root/$v/master"
-    val raw = base("raw")
-    val blob = base("blob")
-
-    val request = new Request(
-      s"https://api.github.com/repos/$organization/$repository/readme",
-      new RequestInit {
-        headers = headersWithCreds.toJSDictionary
-      }
-    )
-    fetch(request).toFuture
-      .flatMap { res =>
-        if (res.status == 200) {
-          res.text().toFuture
-        } else {
-          Future.successful("No README found for this project, please check the repository")
+    def setReadme(): Future[Unit] = {
+      val readmeRequest: Request = new Request(
+        s"https://api.github.com/repos/$organization/$repository/readme",
+        new RequestInit {
+          headers = headersWithCreds.toJSDictionary
         }
-      }
-      .foreach { res =>
-        element.innerHTML = res
+      )
+
+      fetch(readmeRequest).toFuture
+        .flatMap { res =>
+          if (res.status == 200) {
+            res.text().toFuture
+          } else {
+            Future.successful("No README found for this project, please check the repository")
+          }
+        }
+        .flatMap(res => Future { element.innerHTML = res })
+    }
+
+    def setLogo(): Future[Unit] = {
+      def getLogoAndSetHTML(branch: String, organization: String, repository: String): Unit = {
```

Comment:
It does not just set the logo, it fixes all the URLs of all the images in the readme file.

```suggestion
    def getDefaultBranchAndFixImages(): Future[Unit] = {
      def fixImages(branch: String, organization: String, repository: String): Unit = {
```